src/contracts/UpgradeRegressionManager.sol

```diff
@@ -416,4 +420,51 @@ contract UpgradeRegressionManager is IUpgradeRegressionManager {
     emit AdminSet(admin, _newAdmin);
     admin = _newAdmin;
   }
+
+  /// @notice Returns the current state of a proposed rollback.
+  /// @param _rollback The rollback to check.
+  /// @return The current ProposalState of the rollback, which can be:
+  /// - `Unknown`: rollback does not exist (i.e., never proposed).
+  /// - `Pending`: proposed but not yet queued and not expired.
+  /// - `Expired`: proposed but not queued before expiration.
+  /// - `Queued`: queued for execution, but delay not elapsed.
+  /// - `Active`: queued and ready for execution.
+  /// - `Executed`: rollback has already been executed (terminal state).
+  /// - `Canceled`: rollback was canceled before execution (terminal state).
+  /// @dev This function determines the rollback's lifecycle state based on timestamps and flags.
+  function _getState(Rollback memory _rollback) internal view returns (ProposalState) {
+    // Revert if the rollback was not proposed (i.e., does not exist).
+    if (_rollback.queueExpiresAt == 0) {
+      return ProposalState.Unknown;
```

Comment:
The Governor actually [reverts](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/Governor.sol#L158) when the ID is not recognized. What do you think of following that convention and removing this new state from the struct?

Reply:
Yeah, I would agree with your approach if we were building in isolation. If we're going to aim for consistency with the Governor, though, (which I think we should) then it's probably best to stay as close to the Governor as possible.